dbt/models/iasworld/schema/iasworld.pardat.yml

```diff
@@ -167,6 +167,32 @@ sources:
             description: Jurisdictions (Municipalities) used as part of Folio number (BCA)
           - name: nbhd
             description: '{{ doc("shared_column_nbhd_code") }}'
+            tests:
+            - row_values_match_after_join:
+                name: iasworld_pardat_nbhd_matches_legdat_township
+                external_model: source('iasworld', 'legdat')
+                external_column_name: user1
+                column_alias: nbhd_code
+                external_column_alias: township_code
+                additional_select_columns:
+                  - model.taxyr
+                  - model.parid
+                  - model.who
+                  - model.wen
+                join_condition:
+                  ON  substr(model.nbhd, 1, 2) != external_model.user1
+                  AND model.parid = external_model.parid
+                  AND model.taxyr = external_model.taxyr
+                  AND model.cur = 'Y'
+                  AND model.deactivat IS NULL
```

Comment:
**[Nitpick, non-blocking]** It's a little bit arbitrary, but our pattern so far as been to include static filters like these in the `config.where` block rather than the `join_condition`, mostly because it helps us reuse `config.where` logic via the `unique-conditions` anchor. I'm open to being convinced that there's a good reason to do it here rather than in `config.where`, though.

Reply:
There actually is a reason here: conditionals in the `config: where:` clause can't be namespaced to a table since they're applied to the whole test result (which is an inner query). Since columns here share the same name across joined tables, we need to include the namespacing.

Reply:
Hmm, I don't quite understand why namespacing is required in this case. I understand why any filters applied in the `join_condition` block need to be namespaced, but any filters in the `config.where` block are applied to the base model in a subquery prior to the join, so there shouldn't be any namespacing issues as far as filtering the base model is concerned. For example, running the test with this diff returns the same set of results as the current code in this PR:

```diff
--- a/dbt/models/iasworld/schema/iasworld.pardat.yml
+++ b/dbt/models/iasworld/schema/iasworld.pardat.yml
@@ -183,13 +183,9 @@ sources:
                   ON  substr(model.nbhd, 1, 2) != external_model.user1
                   AND model.parid = external_model.parid
                   AND model.taxyr = external_model.taxyr
-                  AND model.cur = 'Y'
-                  AND model.deactivat IS NULL
                   AND external_model.cur = 'Y'
                   AND external_model.deactivat IS NULL
-                config:
-                  where:
-                    CAST(taxyr AS int) BETWEEN {{ var('test_qc_year_start') }} AND {{ var('test_qc_year_end') }}
+                config: *unique-conditions
                 meta:
                   category: relationships
                   description: nbhd code first 2 digits should match legdat.user1 (township code)
```

But maybe I'm misunderstanding something about the way the join is constructed! I don't think it's a particularly big deal either way.

Reply:
Oh I see I see, you only wanted to move the `model.` conditions into the `config:` not both the `model.` and `external_model.` ones. That makes sense. Fixed in 24b97da!